File: templates/Uwp/_composition/_shared/Feature.SuspendAndResume.AddNavigation/Services/SuspendAndResumeService_postaction.cs
//{[{
using System.Reflection;
using Windows.UI.Xaml.Controls;
//}]}
namespace Param_ItemNamespace.Services
{
    internal class SuspendAndResumeService : ActivationHandler<LaunchActivatedEventArgs>
    {
        private const string StateFilename = "SuspendAndResumeState";
        //^^
        //{[{
        // TODO WTS: Subscribe to the OnBackgroundEntering event from your current Page to save the current App data.
        // Only one Page should subscribe to OnBackgroundEntering at a time, as app will navigate to that Page on resume.
        public event EventHandler<OnBackgroundEnteringEventArgs> OnBackgroundEntering;

        // TODO WTS: Subscribe to the OnResuming event from the current Page
        // if you need to refresh online data when the app resumes without being terminated.
        public event EventHandler OnResuming;

        //}]}
        protected override async Task HandleInternalAsync(LaunchActivatedEventArgs args)
        {
            //^^
            //{[{
            if (saveState?.Target != null && typeof(Page).IsAssignableFrom(saveState.Target))
            {
                NavigationService.Navigate(saveState.Target, saveState.SuspensionState);
            }
            //}]}
        }
    }
}

File: templates/Uwp/_composition/_shared/Feature.SuspendAndResume.AddNavigation._VB/Services/SuspendAndResumeService_postaction.vb
'{[{
Imports System.Reflection
'}]}
Namespace Services

    Friend Class SuspendAndResumeService
        Inherits ActivationHandler(Of LaunchActivatedEventArgs)

        Private Const StateFilename As String = "SuspendAndResumeState"

        '{[{
        ' TODO WTS: Subscribe to the OnBackgroundEntering event from your current Page to save the current App data.
        ' Only one Page should subscribe to OnBackgroundEntering at a time, as the App will navigate to that Page on resume.
        Public Event OnBackgroundEntering As EventHandler(Of OnBackgroundEnteringEventArgs)

        ' TODO WTS: Subscribe to the OnResuming event from the current Page
        ' if you need to refresh online data when the App resumes without being terminated.
        Public Event OnResuming As EventHandler
        '}]}

        Protected Overrides Async Function HandleInternalAsync(args As LaunchActivatedEventArgs) As Task
            '^^
            '{[{
            If saveState?.Target IsNot Nothing AndAlso GetType(Page).IsAssignableFrom(saveState.Target) Then
                NavigationService.Navigate(saveState.Target, saveState.SuspensionState)
            End If
            '}]}
        End Function
    End Class
End Namespace
